From ec77aabba7f107b39f4cd0510d279eae752e16c6 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sun, 27 Jan 2019 16:06:42 -0500
Subject: [PATCH 1/3] fastboot boot: gives enough space to fit a pointer

%p on 64 bits platforms can extend over the length of the string
allocated for passing to `bootm`, thus giving 0x00000000 instead of the
expected addresses.
---
 drivers/fastboot/fb_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/fastboot/fb_common.c b/drivers/fastboot/fb_common.c
index c6e06aab7a..17eca73be0 100644
--- a/drivers/fastboot/fb_common.c
+++ b/drivers/fastboot/fb_common.c
@@ -119,7 +119,7 @@ void fastboot_boot(void)
 	if (s) {
 		run_command(s, CMD_FLAG_ENV);
 	} else {
-		static char boot_addr_start[12];
+		static char boot_addr_start[20];
 		static char *const bootm_args[] = {
 			"bootm", boot_addr_start, NULL
 		};
-- 
2.18.1

